state-sync/src/coordinator.rs

```diff
@@ -1103,22 +1097,37 @@ impl<T: ExecutorProxyTrait> StateSyncCoordinator<T> {
             self.local_state.trusted_epoch()
         };
         self.local_state.verify_ledger_info(&response_li)?;
-        if let Some(li) = pending_li {
-            if li != response_li {
-                self.local_state.verify_ledger_info(&li)?;
+        if let Some(new_highest_li) = new_highest_li.clone() {
+            if new_highest_li != response_li {
+                self.local_state.verify_ledger_info(&new_highest_li)?;
             }
```

Comment:
and here we can check if we already have a highest set and if we do we evaluate if this would bump the counter and bump the counter if it would.

Reply:
To clarify, you're concerned about bumping the counter to log the highest that we've seen (not just the highest we've currently pegged too)?

Reply:
Yes, this is about tracking the counter, I think this is where we should do it.

Reply:
Done, thanks!